### DISCUSSION TITLE
The image could not be loaded.

### DISCUSSION THREAD

**Discussion by nzpb** [photoprism/photoprism#3563] [Q&A] [answered]
PLEASE PROCEED ONLY IF YOU ARE ABSOLUTELY SURE THAT THIS IS NOT A TECHNICAL SUPPORT INCIDENT AND/OR POSSIBLY A PROBLEM WITH SOME OTHER SOFTWARE YOU ARE USING. VISIT <https://www.photoprism.app/kb/getting-support> TO LEARN MORE ABOUT OUR SUPPORT OPTIONS. THANK YOU FOR YOUR CAREFUL CONSIDERATION!

---------------------------------------------------------------------------

#### 1. What is not working as documented?

When I open some images, the image could be loaded with an error: The image could not be loaded.

The thumbnail is generated correctly.

#### 2. How can we reproduce it?

1. Go to '.Album or Folders.'
2. Click on 'image thumb'
3. Open the image 
4. See error:The image could not be loaded. (black background with string in white)

![NOTLOADERD](https://github.com/photoprism/photoprism/assets/106904472/39728dc4-2cc4-4327-b6bc-142271b0a438)

When reporting an import, indexing, or performance issue, please include the number and type of pictures in your library, as well as any configuration options you have changed, such as for thumbnail quality.

#### 3. What behavior do you expect?

The error only occur in some pictures but the others are correct. 

![999](https://github.com/photoprism/photoprism/assets/106904472/4a43b4dd-064a-4955-9256-dcb931408ee1)

#### 4. What could be the cause of your problem?

Always try to determine the cause of your problem using the checklists at <https://docs.photoprism.app/getting-started/troubleshooting/> before submitting a bug report.

#### 5. Can you provide us with example files for testing, error logs, or screenshots?

There is no error in libaray - logs when the error occured.

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: AMD64, ARM64, ARMv7,...

Build [230625-17242fb07](https://docs.photoprism.app/release-notes/)   docker - compose , Ubuntu. with "    image: photoprism/photoprism:latest" 

(b) Database Type & Version: MariaDB,  

(c) Operating System Types & Versions: Linux, Windows, Android,...

Ubuntu 22 / docker 

(d) Browser Types & Versions: Firefox, Chrome, Safari on iPhone,...

I tried all of browsers in Chrome  ff and phones.

(e) Ad Blockers, Browser Plugins, and/or Firewall Software?

no, use different client browsers, still have this problem.

You can find the version/build number of the app in *Settings* by scrolling to the bottom. Note that MySQL 8 support has been discontinued, see system requirements at <https://docs.photoprism.app/getting-started/#system-requirements>.

*Always provide database and operating system details if it is a backend, import, or indexing issue. Should it be a frontend issue, at a minimum we require you to provide web browser and operating system details. When reporting a performance problem, we ask that you provide us with complete information about your environment, as there may be more than one cause.*

#### 7. On what kind of device is PhotoPrism installed?

This is especially important if you are reporting a performance, import, or indexing issue. You can skip this if you're reporting a problem you found in our public demo, or if it's a completely unrelated issue, such as incorrect page layout.

(a) Device / Processor Type: Raspberry Pi 4, Intel Core i7-3770, AMD Ryzen 7 3800X,...

I5-8500, VM, Ubuntu 

(b) Physical Memory & Swap Space in GB

6GB  

(c) Storage Type:  SSD 

(d) Anything else that might be helpful to know?

*Always provide device, memory, and storage details if you have a backend, performance, import, or indexing issue.*

#### 8. Do you use a Reverse Proxy, Firewall, VPN, or CDN?

NO, I tried in LAN with Local address, no ssl or nginx used.

If yes, please specify type and version. You can skip this if you are reporting a completely unrelated issue.

*Always provide this information when you have a reliability, performance, or frontend problem, such as failed uploads, connection errors, broken thumbnails, or video playback issues.*

**Using NGINX?** Please also provide the configuration and/or consider asking the NGINX community for advice as we do not specialize in supporting their product. Docs can be found at <https://docs.photoprism.app/getting-started/proxies/nginx/>.

**Comment by graciousgrey** [comment]
It does not seem to be a general bug as no one else is experiencing this. Could you send us a sample file for testing to samples@photoprism.app ?

**Comment by fl0wm0ti0n** [comment]
i run into the same behavior...

**Reply by lastzero** [reply]
Did you already go through the troubleshooting checklists we provide? See for example:
- https://docs.photoprism.app/getting-started/troubleshooting/#broken-thumbnails

**Reply by fl0wm0ti0n** [reply]
Thanks, but the thumbnails are not broken.. the images behind the thumbnails couldn't be shown...
not all but alot of... there is no pattern..

if i want to show the files over "https://domain/library/index/files/  there are the most files missing... but over places and search i have all thumbnails...

i get no errors while indexing... or in the container logs

**Reply by lastzero** [reply]
Like here? https://demo.photoprism.app/library/index/files/2002

That's to be expected as this is a file browser, not a photo browser, so not all items have a preview.

**Reply by fl0wm0ti0n** [reply]
no, not like that. thumbnails are shown or the folders are also here, but i miss the files in the folder. 

and in the other views like search, places etc. there i miss the pictures behind the thumbnails, so if I click on a thumbnail it shows "the image could not be loaded" but not all images.. the error is especially with new pictures

**Reply by lastzero** [reply]
Might be some kind of config issue then, but it's not possible to tell you something specific without knowing more....

**Reply by fl0wm0ti0n** [reply]
i went thru the checklist did some steps like the swapfile to 4gb
i indexed all files again. still same behavior, and i dont see a difference in comparison to pictures which can be showed

i have alot of these warnings:

metadata: unknown json in 2ed079926b666cfd7834272e94548a163d8e109f_exiftool.json
2023-11-21 10:00:58
metadata: unknown json in 639634ab67a08d0acd3922efcdd67d0a3d996220_exiftool.json
2023-11-21 10:00:57

what do they say?

**Reply by fl0wm0ti0n** [reply]
its strange, i can also download the picture behind the thumbnail. just showing doesnt work...

**Reply by graciousgrey** [reply]
Do you use a proxy or any browser plugins? Did you test another browser?

**Reply by fl0wm0ti0n** [reply]
but why then just most of new pictures are affected? so many pictures are working.

tried chrome and firefox

**Reply by lastzero** [reply]
Just to be sure: do you use our official Docker image? What specific version?

**Reply by fl0wm0ti0n** [reply]
image: photoprism/photoprism:latest

[PhotoPrism® CE](https://look.omniflow.cc/library/about)
Build [230923-e59851350](https://docs.photoprism.app/release-notes/)

**Reply by lastzero** [reply]
I'd recommend updating and performing a full rescan before you try anything else:
- https://docs.photoprism.app/release-notes/#october-21-2023
- https://docs.photoprism.app/getting-started/updates/

**Reply by graciousgrey** [reply]
If updating and reindexing does not help. Please check whether the problem persists with the same photos and different window sizes or on the phone. 
It would also be good to know which thumbnails exist for these photos.  You can find the thumbnails in storage/cache/thumbnails, sorted in folders according to the hash of the photos. The hash can be found in the photo editing dialog.

**Reply by fl0wm0ti0n** [reply]
i updated it to latest, and reindexted all. problem persists :(

problem is on phone, and a few tested browsers. 

as i can download the pictures behind the thumbnail i don't assume it cant find the correct picture. it just cant be shown :(

i give your thumbnails idea a try, lets see

**Reply by lastzero** [reply]
No proxy, cdn, or vpn in use?

**Reply by fl0wm0ti0n** [reply]
no, whats a cdn?, and no

and how would that effect just a few photos?

**Reply by lastzero** [reply]
CloudFlare for example....

**Reply by lastzero** [reply]
Caching / wrong Caching obviously can affect it. Also make sure that you have dynamic thumbnail rendering enabled in Settings  > Advanced so that missing files can be created on demand when needed.

**Reply by fl0wm0ti0n** [reply]
The setting you said was activated

is it normal that after reindexing with the newest version the used space is more then double the size`? could it be the data is saved dublicate?

My storage is full now for around 70.000 Pictures and Videos it uses:
VOLUME NAME                                                        LINKS     SIZE
photoprism_storage_data                                            1         361.3GB
services_mariadb_data                                              1         860.8MB

**Reply by lastzero** [reply]
Nope. Either its a config issue or the space is used by transcoded videos.

**Reply by fl0wm0ti0n** [reply]
maybe i should delete everything, also the DB? and start new?

**Reply by fl0wm0ti0n** [reply, marked as answer]
well after deleting everything and recreating the container with new db and volumes everything works fine...

**Comment by ianfixes** [comment]
Is there a more targeted method of fully destroying the app container and rebuilding it?  I am experiencing this issue, but not running the app from a container.  I have run a forced re-indexing of my entire library and this does not fix the missing images.  

If there is a bug/issue I can subscribe to as well, that would be useful.

**Reply by lastzero** [reply]
Did you already try running the `photoprism thumbs -f` command in a terminal? See https://docs.photoprism.app/getting-started/docker-compose/#examples for additional command examples.

**Reply by ianfixes** [reply]
I had not been using docker compose -- I had followed [a guide to install directly on Debian](https://github.com/IndrekHaav/photoprism-debian).  Since that guide doesn't create a `photoprism` user with shell access, I couldn't get that command to work.  

Long story short, I switched to the `docker-compose` method and just rebuilt everything from scratch, and it works fine.  But this time I should be in better shape to work with any of the maintenance commands.

**Comment by anst84** [comment]
I got the same issue. Is "deleting everything and recreating the container with new db and volumes everything works fine..." the only option to make it work? Are there any guides available for doing this operation? My installation is quite fresh for what it is worth...

**Reply by lastzero** [reply]
Have you already tried re-indexing and going through the troubleshooting checklists we provide?
:point_right:  https://docs.photoprism.app/getting-started/troubleshooting/#broken-thumbnails

**Comment by vadimzharov** [comment]
Hi! I've got exactly the same issue - some images are not displaying (with error message "the image could not be loaded"), but all thumbnails are in place. All images are stored in one place (NFS share), and displayed correctly if I'm using 3rd party tool.
There are no any errors in the photoprism - for **all** requested images (the displayed ones and the ones not showing up) the status code is 200:
```
time="2024-04-24T14:11:53Z" level=debug msg="server: GET /api/v1/t/17d1076ac6aeff9685dc1e13a7f0f05a78e5d266/255xoi46/fit_1920 (200) [58.813784ms]"
time="2024-04-24T14:12:11Z" level=debug msg="server: GET /api/v1/t/abb60861c1f1d8735587300e624aa478681c7e3f/255xoi46/fit_2048 (200) [1.514689ms]"
```

If I click on "The image" link, the new tab opens tying to load the image but showing blue questioning mark icon (the URL is https://xxx.com/api/v1/t/abb60861c1f1d8735587300e624aa478681c7e3f/255xoi46/fit_2048)

I'm observing this behavior in all browsers, however I found the reason why this error appears. 
I'm using Macbook, and if the browser window (no matter what browser is - Chrome/Firefox) is in Macbook display - then I'm getting the error "The image could not be loaded", **but** if I move the same browser window to my external display - then the error is gone, and all images are displaying correctly.
I checked the image URL Photoprism is generating:
If the browser window is in Macbook display, then the image URL is https://XXX.com/api/v1/t/abb60861c1f1d8735587300e624aa478681c7e3f/255xoi46/fit_2048 (ends with fit_2048), and the error appears ("The image could not be loaded").
But if I move browser window to external display - then the image URL becomes:
https://XXX/api/v1/t/abb60861c1f1d8735587300e624aa478681c7e3f/255xoi46/fit_1920 (ends with fit_1920), and all images are displayed (not error for the same image).

If I move browser window to Macbook display, where image is not showing and Photoprism generates URL ending with fit_2048, but use URL ending with fit_1920 - the image is showing correctly (just by pasting the URL https://XXX.com/api/v1/t/abb60861c1f1d8735587300e624aa478681c7e3f/255xoi46/fit_1920).

Also, what I noticed is that if I'm checking with developer tools when I click on the image in Photoprism, then:
If URL ends with fit_2048 (https://xxx.com/api/v1/t/abb60861c1f1d8735587300e624aa478681c7e3f/255xoi46/fit_2048) - then browser is loading only two files - fit_2048 and favicon.ico
But if URL ends with fit 1920 (https://XXX.com/api/v1/t/abb60861c1f1d8735587300e624aa478681c7e3f/255xoi46/fit_1920) - then browser is loading four files - fit_1920, favicon.ico, content.js, executor.js.

Is this a Photoprism error or something related to browsers/OS?

**Reply by lastzero** [reply]
The viewer considers the display pixel density (eg regular / retina display) when calculating the size of thumbnails being loaded. So it could be related to this? Since the pixel density isn't dynamically updated in the code as you move your browser window, I'd expect that you need to reload the user interface - if browsers can property handle this situation at all.

**Reply by vadimzharov** [reply]
What do you mean by reload the user interface - reload browser window (ctrl+r)? I tried, it doesn't help. Also, I tried to disconnect external display, restart browser - also didn't help.

**Reply by lastzero** [reply]
But it works when you don't have mixed display types? Do you see any errors or warnings in the browser log (JavaScript console)?

**Reply by vadimzharov** [reply]
No, actually I noticed the error on Macbook without any displays attached. Then I started to debug this on another MacBook with attached external display and found that behavior depending on the display type.
Interestingly, if browser window in on Macbook display and I open and extend developer tools so it takes half of the browser window in Chrome, then all images are displaying correctly because Photoprism generates URLs ending with fit_1920:
<img width="1505" alt="image" src="https://github.com/photoprism/photoprism/assets/23321919/0f4a6fb8-e222-4b01-8d33-d587cc6f6076">

If I reproduce the error - no errors in the console:
<img width="1505" alt="image" src="https://github.com/photoprism/photoprism/assets/23321919/1b1a209e-040e-48da-a8c1-399842ea7831">

**Reply by lastzero** [reply]
If it depends on the browser window size, you should enable dynamic previews (thumbnails) as the cached sized may be too small:
https://demo.photoprism.app/library/settings/advanced

**Reply by vadimzharov** [reply]
It had been enabled once I installed Photoprism, so a while ago. These are my settings:
<img width="1230" alt="image" src="https://github.com/photoprism/photoprism/assets/23321919/e433bb8f-6ace-4a85-8248-ddbb906e9d8e">

**Reply by lastzero** [reply]
Do you have browser plugins or an ad blocker of any kind?

**Reply by vadimzharov** [reply]
This error appears on all my browsers - Safari - no plugin/ad blocker, Chrome - ublock installed, Firefox - fresh install, no plugins/adblockers

**Reply by lastzero** [reply]
Maybe the cached file is broken for some reason? Did you try regenerating it with the photoprism thumbs command in a terminal?

**Reply by vadimzharov** [reply]
Ok, so it seems regenerating cache files helped. In my case (I store many photos) it took about 24 hours.
So after executing command
```
photoprism thumbs -f
```
and waited for the command to complete all images are displaying correctly. 
No need to delete/recreate DB and volumes.

**Reply by lastzero** [reply]
Could be that there were broken thumbs because the process crashed previously, for example when a memory limit was hit or the computer / Docker was restarted.

**Reply by vadimzharov** [reply]
Is there a way to regenerate only broken thumbs?

**Reply by lastzero** [reply]
Depends. A pull request for only regenerating thumbs with empty files (size 0) should be easy to create. Beyond that, we'd first need to figure out why they were broken and how to recognize that.

**Reply by blusius** [reply]
I also faced loading issue. Found that on windows laptop the same images was opening just fine. But on IPad and mac was failing to load no matter what browser used.

My PhotoPrism (Build 250228-43447fa38) is inside proxmox lxc. Durring scan I have several issues, storage shortage, memory shortage, stoped (not shutdown) myself to increase memory and so on. I gues it is related.

`photoprism thumbs -f ` (20 min.) DID NOT FIX my issue. 
Restarted prism, did complete rescan on one of the folders, cleared browser cache. Enabled "Dynamic Previews". No sucess.

I still see loaded empty images in browsers network tab. 
![image](https://github.com/user-attachments/assets/74809477-aa6e-4481-9af4-51b73136c2aa)


Recreating everything sounds silly. This could happen anytime, when there is no sulution to fix to loose all time spent on tagging and face managing is not an option.

Any more ideas?

**Reply by blusius** [reply]
noticed that `photoprism thumbs -f` regenerates only for images extracted from video files. I have no such files named .mov.jpg or mp4.jpg in my collection.  And usual images are skipped somehow...

![image](https://github.com/user-attachments/assets/1067660d-eb44-4b21-8c45-c6e4cd9e71bb)

**Reply by graciousgrey** [reply]
The thumbs -f command should also create thumbnails for jpg and png files. I would recommend to locate the thumbnails that are not loading in the cache folder to see if they have been re-created by the thumbs command. Files in the cache folder are sorted by the file hash (you can find the file hash on the files tab of the edit dialog).
If the thumbs could not be created the reason could  be permission issues or a full storage folder. Do you see any warning or errors on the error page?
https://docs.photoprism.app/getting-started/troubleshooting/#broken-thumbnails
